pkg/codegen/convertutil/utilities.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2016-2020, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package convertutil
+
+import (
+	"github.com/blang/semver"
+
+	"github.com/pulumi/pulumi/pkg/v3/codegen/hcl2/model"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+)
+
+func SetHighestPackageVersion(versionExpr *model.Expression, pkg string, packages []*schema.Package) {
+	if tExpr, ok := (*versionExpr).(*model.TemplateExpression); ok {
```

Comment:
From the function signature, it looks like this conversion should never fail. Can we remove the check? If not, should we add a reporting mechanism when `SetHighestPackageVersion` is used incorrectly (and the check fails)?

Reply:
True, this should always work here.